src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -423,6 +419,15 @@ public void updateReplicateStatus(OptimisticLockingFailureException exception,
                                                                 String walletAddress,
                                                                 ReplicateStatusUpdate statusUpdate,
                                                                 UpdateReplicateStatusArgs updateReplicateStatusArgs) {
+        // Synchronization is mandatory there to avoid race conditions.
+        // Lock key should be unique, e.g. `chainTaskId + walletAddress`.
+        return replicatesUpdateLockRunner.getWithLock(
+                chainTaskId + walletAddress,
```

Comment:
```suggestion
        String lockKey = chainTaskId + walletAddress;
        return replicatesUpdateLockRunner.getWithLock(
                key,
```

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-core/pull/492/commits/6361f1cb2658065355f4fac4edd2f7ce5452055a.